File: algorithms/valid-parentheses/ValidParentheses.java
import java.util.Stack;

public class ValidParentheses {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        for (char chars : s.toCharArray()) {
            if (chars == '(' || chars == '[' || chars == '{') {
                stack.push(chars);
            } else {
                if (stack.isEmpty()) {
                    return false;
                }
                char top = stack.peek();
                if ( (chars == ')' && top == '(') || (chars == ']' && top == '[') || (chars == '}' && top == '{')) {
                    stack.pop();
                } else {
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }

    public static void main(String[] args) {
        String s = "(){}";
        boolean result = new ValidParentheses().isValid(s);
        System.out.println(s + " = " + result);
    }
}

File: algorithms/valid-parentheses/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for chars in list(s):
            if chars == '(' or chars == '[' or chars == '{':
                stack.append(chars)
            else:
                if self.isEmpty(stack):
                    return False
                top = self.peek(stack)
                if (top == '(' and chars == ')') or (top == '[' and chars == ']') or (top == '{' and chars == '}'):
                    stack.pop()
                else:
                    return False
        return self.isEmpty(stack)

    def isEmpty(self, stack) -> bool:
        if stack == []:
            return True
        else:
            return False
    
    def peek(self, stack):
        if stack == []:
            return None
        else:
            top = len(stack) -1
            return stack[top]
        
s = "(){}[]"
result = Solution().isValid(s)
print(s, " -> ", result)